.github/workflows/typecheck.yml

```diff
@@ -24,8 +24,16 @@ jobs:
       - name: Check for new JavaScript files
         run: |
           git fetch origin main --no-tags --depth=1
-          count_new_js=$(git diff --name-only --diff-filter=A origin/main HEAD -- 'src/libs/*.js' 'src/hooks/*.js' 'src/styles/*.js' 'src/languages/*.js' | wc -l)
+
+          # Explanation:
+          #  - comm is used to get the intersection between two bash arrays
+          #    - git diff is used to see the files that were added on this branch
+          #    - gh pr view is used to list files touched by this PR. Git diff may give false positives if the branch isn't up-to-date with main
+          #  - wc counts the words in the result of the intersection
+          count_new_js=$(comm -1 -2 <(git diff --name-only --diff-filter=A origin/main HEAD -- 'src/libs/*.js' 'src/hooks/*.js' 'src/styles/*.js' 'src/languages/*.js') <(gh pr view ${{ github.event.pull_request.number }} --json files | jq -r '.files | map(.path) | .[]') | wc -l)
```

Comment:
ah I see so this is the meat

so this line is counting how many new JavaScript files (in a specified directoreis) have been added in the current branch compared to the main branch? and you're using `gh pr view` because `git diff` might compare the current branch with the outdated main? And then you're making sure the newly added JS files are zero.

If this is the case, the logic seems correct. LGTM

Reply:
Yep, that's all correct.